artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueuePendingMessageMetrics.java

```diff
@@ -67,6 +74,7 @@ public void incrementMetrics(final MessageReference reference) {
    public void decrementMetrics(final MessageReference reference) {
       long size = -getPersistentSize(reference);
       COUNT_UPDATER.decrementAndGet(this);
+      logger.debugf("%s decrement messageCount to %d: %s", this, messageCount, reference);
```

Comment:
can we put inside if debug enabled (as on hot path) so will be removed by jit if debug isnt on.

Reply:
Will fix.